src/storage/temporary_file_manager.cpp

```diff
@@ -165,8 +194,13 @@ TemporaryDirectoryHandle::TemporaryDirectoryHandle(DatabaseInstance &db, string
 	auto &fs = FileSystem::GetFileSystem(db);
 	if (!temp_directory.empty()) {
 		if (!fs.DirectoryExists(temp_directory)) {
+			auto &config = DBConfig::GetConfig(db);
 			fs.CreateDirectory(temp_directory);
 			created_directory = true;
+			// Maximum swap space isn't set explicitly, initialize to default
+			if (!config.options.maximum_swap_space.ExplicitlySet()) {
+				config.SetDefaultMaxSwapSpace(&db);
```

Comment:
Instead of modifying the DBConfig here - can we store this setting in the `TemporaryDirectoryHandle` directly? Then we no longer need to differentiate between "user set" and "system set" - any setting in the DBConfig is always user set.